WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs

```diff
@@ -1321,357 +1327,332 @@ private async Task<JObject> HandleMergeBranchActionAsync(DataRow dataRowWithSett
+                            actionData.ItemIdOriginal = String.Equals(actionData.OldValue, "item_id", StringComparison.OrdinalIgnoreCase) ? Convert.ToUInt64(actionData.NewValue) : 0;
+                            actionData.LinkIdOriginal = String.Equals(actionData.OldValue, "itemlink_id", StringComparison.OrdinalIgnoreCase) ? Convert.ToUInt64(actionData.NewValue) : 0;
+                            columnNameForFileLink = actionData.OldValue;
+                            actionData.OldValue = null;
+                            actionData.NewValue = null;
+
+                            actionData.ItemIdMapped = actionData.ItemIdOriginal;
+                            actionData.LinkIdMapped = actionData.LinkIdOriginal;
+
+                            // If we have a table prefix, then we need to check if it's a valid link type.
+                            if (Int32.TryParse(tablePrefix.TrimEnd('_'), out var linkTypeFromTablePrefix) && allLinkTypeSettings.Any(t => t.UseDedicatedTable && t.Type == linkTypeFromTablePrefix))
+                            {
+                                actionData.LinkType = linkTypeFromTablePrefix;
+                            }
+
+                            var fileData = await GetFileDataAsync(actionData.FileIdOriginal, sqlParameters, tableName, tablePrefix, branchConnection, filesCache, actionData);
+                            if (fileData.IsDeleted)
                             {
-                                linkCacheData = await GetLinkDataAsync(linkId, sqlParameters, tableName, branchConnection, linksCache);
+                                // If the file is deleted, we can use the item id and link id from wiser_history.
+                                fileData.ItemId = actionData.ItemIdOriginal;
+                                fileData.LinkId = actionData.LinkIdOriginal;
+                            }
+
+                            // If the file is linked to a link, then we need to find the data of that link.
+                            if (actionData.LinkIdOriginal > 0)
+                            {
+                                actionData.LinkTableName = tableName.Replace(WiserTableNames.WiserItemFile, WiserTableNames.WiserItemLink, StringComparison.OrdinalIgnoreCase);
+                                linkCacheData = await GetLinkDataAsync(actionData.LinkIdOriginal, sqlParameters, actionData.LinkTableName, branchConnection, linksCache, actionData);
+                                actionData.ItemIdOriginal = linkCacheData.ItemId ?? 0;
+                                actionData.ItemIdMapped = linkCacheData.ItemId ?? 0;
+                                actionData.LinkDestinationItemIdOriginal = linkCacheData.DestinationItemId ?? 0;
+                                actionData.LinkDestinationItemIdMapped = linkCacheData.DestinationItemId ?? 0;
```

Comment:
Is it correct both properties use the same value?

Reply:
Yes that is correct; We want to make sure that we always have a mapped ID. So we always fill the original ID there first, then later we check if that has a mapped ID and if yes, we change it. If it doesn't have a mapped ID, then it will stay the same as the original ID. This way we can just use the mapped ID whenever we need to use that ID in the production database and we're sure it always has a value. See lines 1720+ in this class where we overwrite these values with the actual mapped IDs, if they exist.